dll/win32/kernel32/client/utils.c

```diff
@@ -372,28 +377,36 @@ BaseCreateStack(HANDLE hProcess,
     Headers = RtlImageNtHeader(NtCurrentPeb()->ImageBaseAddress);
     if (!Headers) return STATUS_INVALID_IMAGE_FORMAT;
 
-    StackCommitHeader = Headers->OptionalHeader.SizeOfStackCommit;
-    StackReserveHeader = Headers->OptionalHeader.SizeOfStackReserve;
-
-    if (!StackReserve) StackReserve = StackReserveHeader;
+    if (StackReserve == 0)
+        StackReserve = Headers->OptionalHeader.SizeOfStackReserve;
 
-    if (!StackCommit)
+    if (StackCommit == 0)
     {
-        StackCommit = StackCommitHeader;
+        StackCommit = Headers->OptionalHeader.SizeOfStackCommit;
     }
     else if (StackCommit >= StackReserve)
     {
+        /* Grow the reserve beyond the commit, up to 1MB alignment */
         StackReserve = ROUND_UP(StackCommit, 1024 * 1024);
     }
 
+    /* Align everything to Page Size */
     StackCommit = ROUND_UP(StackCommit, PageSize);
     StackReserve = ROUND_UP(StackReserve, AllocationGranularity);
 
+#if 0
     GuaranteedStackCommit = NtCurrentTeb()->GuaranteedStackBytes;
     if ((GuaranteedStackCommit) && (StackCommit < GuaranteedStackCommit))
     {
         StackCommit = GuaranteedStackCommit;
     }
+#else
```

Comment:
Why?

Reply:
Fixed; GuaranteedStackBytes feature cannot be used when the thread (and thus the thread stack) is being created (it would always be zero at that point); instead the MinimumStackSize has to be used. It is manually through the call to SetThreadStackGuarantee() that one can set a non-zero value to GuaranteedStackBytes (and at the same time enlarge the stack if needed).